app/lib/actions/katello/content_view_version/incremental_update.rb

```diff
@@ -358,6 +358,18 @@ def copy_deb_content(new_repo, dep_solve, deb_ids)
         def copy_yum_content(new_repo, dep_solve, package_ids, errata_ids)
           copy_outputs = []
           if new_repo.content_type == ::Katello::Repository::YUM_TYPE
+            library_instance = new_repo.library_instance
+            content_present_in_this_repo = library_instance
+                                           .rpms
+                                           .with_identifiers(package_ids)
+                                           .exists?
+
+            content_present_in_this_repo ||= library_instance
+                                              .errata
+                                              .with_identifiers(errata_ids)
+                                              .exists?
+            return [] unless content_present_in_this_repo
+
             unless errata_ids.blank?
```

Comment:
Should the new checks you added happen within the blocks of these `.blank?` checks below? We can avoid some DB queries that way, it seems.

Reply:
I mean 
```ruby
 library_instance
       .rpms
       .with_identifiers(package_ids)
       .exists?
```
results in exactly 1 query  , but I can see happening in the blank the blocks being a good idea. Going to implement that